librad/src/git/storage.rs

```diff
@@ -485,6 +496,62 @@ impl Storage {
         }))
     }
 
+    pub fn entity_blob<'a>(&'a self, commit: git2::Commit<'a>) -> Result<git2::Blob, Error> {
+        blob(&self, commit.tree()?, "id")
+    }
+
+    pub fn entity_parent_commit(&self, commit_oid: git2::Oid) -> Option<git2::Commit> {
+        self.backend
+            .find_commit(commit_oid)
+            .ok()
+            .and_then(|commit| match commit.parents().len() {
+                1 => commit.parents().next(),
+                _ => None,
+            })
+    }
+
+    pub fn linear_commit_history<'a>(
+        &'a self,
+        commit: git2::Commit<'a>,
+    ) -> LinearHistoryCommits<'a> {
+        LinearHistoryCommits(Some(commit))
+    }
+
+    pub fn entity_metadata_commit(&self, urn: &RadUrn) -> Result<git2::Commit, Error> {
```

Comment:
Could be simplified to `self.find_reference(&git::types::Reference::rad_id(urn.id.clone()).to_string())`

Reply:
Also probably not public

Reply:
I did the simplification.
About dubious public APIs, I am making them private and we'll see what's really needed later.